projects/solar/src/solar_fsm.c

```diff
@@ -0,0 +1,92 @@
+#include "solar_fsm.h"
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#include "can.h"
+#include "can_msg_defs.h"
+#include "can_unpack.h"
+#include "drv120_relay.h"
+#include "event_queue.h"
+#include "exported_enums.h"
+#include "fsm.h"
+#include "solar_events.h"
+#include "status.h"
+
+FSM_DECLARE_STATE(state_relay_open);
+FSM_DECLARE_STATE(state_relay_closed);
+
+static bool prv_relay_open_fault_guard(const Fsm *fsm, const Event *e, void *context) {
+  SolarFsmStorage *storage = context;
+  EESolarFault fault = GET_FAULT_FROM_EVENT(*e);
+  for (uint8_t i = 0; i < storage->num_relay_open_faults; i++) {
```

Comment:
Seems pretty possible that non-fault event could be parsed as a fault event, no?

Also, I feel like this logic should go in "fault_tx" (with it being renamed probably) because it already catches these events. As I mentioned before ... you can probably just call a function i.e. `raise_fault` which does both the can transmit and if necessary calls a function here to advance the FSM.